src/modules/adaptors/Convex/ConvexAdaptor.sol

```diff
@@ -0,0 +1,268 @@
+     * @dev Booster.sol serves as the primary contract that accounts for markets via poolIds. PoolInfo structs can be queried w/ poolIds, where baseRewardPool contracts, and other info can be obtained.
+     */
+    constructor(address _booster) {
+        booster = IBooster(_booster);
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Convex Curve Adaptor V 0.1"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice Deposit & Stakes LPT from the cellar into Convex Market at the end of the user deposit sequence.
+     * @param assets amount of LPT to deposit and stake
+     * @param adaptorData see adaptorData info at top of this smart contract
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        (uint256 pid, address rewardsPool) = abi.decode(adaptorData, (uint256, address));
+        _validatePositionIsUsed(pid, rewardsPool);
+        (address lpToken, , , , , ) = booster.poolInfo(pid);
+        ERC20 lpt = ERC20(lpToken); // TODO: double check that struct object is coming out of this properly
+        lpt.safeApprove(address(booster), assets);
+        booster.deposit(pid, assets, true);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(lpt, address(booster));
+    }
+
+    /**
+     * @notice If a user withdraw needs more LPTs than what is in the Cellar's wallet, then the Cellar will unstake, unwrap cvxLPTs, and withdraw LPTs from Convex
+     * @param amount of LPT to unstake, unwrap, and withdraw from Convex market
+     * @param receiver see baseAdaptor.sol
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @param configurationData see configurationData at top of this smart contract
+     */
+    function withdraw(
+        uint256 amount,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public override {
+        // Check that position is setup to be liquid.
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (!isLiquid) revert BaseAdaptor__UserWithdrawsNotAllowed();
+
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        (uint256 pid, address rewardPool) = abi.decode(adaptorData, (uint256, address));
+        _validatePositionIsUsed(pid, rewardPool);
```

Comment:
Not sure if we really need this check here since we are only pulling money on a withdraw, vs sending money else where. I guess go with whatever convention we have used in the past, like did we do a isPositionUsed check for compound withdraws? Or for Aura/ERC4626 withdraws?

Reply:
Within Aura/ERC4626 withdraws we have:

```
        _verifyERC4626PositionIsUsed(address(erc4626Vault));
        erc4626Vault.withdraw(assets, receiver, address(this));
```

and within Compound we have:

```
        CErc20 cToken = abi.decode(adaptorData, (CErc20));
        _validateMarketInput(address(cToken));
```

but I see what you're saying. Hmm, perhaps we just keep it and see if there are any comments that come up within the external audit for this?